simulator/infra/cache/t/miss_rate_sim.cpp

```diff
@@ -1,115 +0,0 @@
-/**
- * miss_rate_sim.cpp
- * Program for checking performance of different cache configurations.
- * MIPT-MIPS Assignment 5.
- * Ladin Oleg.
- */
-
-/* C libraries. */
-#include <cstdlib>
-
-/* C++ libraries. */
-#include <iostream>
-#include <fstream>
-
-/* Simulator modules. */
-#include "../cache_tag_array.h"
-
-int main( int argc, char* argv[])
-{
-    /* Check arguments. */
-    if ( argc != 3)
-    {
-        std::cerr << "ERROR: Wrong number of arguments! Required 2: name of file "
-                  << "with memory access addresses; name of output file with miss "
-                  << "rates." << std::endl;
-        exit( EXIT_FAILURE);
-    }
-    if ( argv[ 1] == nullptr)
-    {
-        std::cerr << "ERROR: Wrong first argument! Required name of file with "
-                  << "memory access addresses." << std::endl;
-        exit( EXIT_FAILURE);
-    }
-    if ( argv[ 2] == nullptr)
-    {
-        srd::cerr << "ERROR: Wrong second argument! Required name of output file "
-                  << "with miss rates." << std::endl;
-        exit( EXIT_FAILURE);
-    }
-
-    /* Open and check input and output files. */
-    std::ifstream file_in;
-    file_in.open( argv[ 1], std::ifstream::in);
-    if ( !file_in.is_open())
-    {
-        std::cerr << "ERROR: Can't open the input file!" << std::endl;
-        exit( EXIT_FAILURE);
-    }
-    std::ofstream file_out;
-    file_out.open( argv[ 2], std::ofstream::out);
-    if ( !file_out.is_open())
-    {
-        std::cerr << "ERROR: Can't open the output file!" << std::endl;
-        exit( EXIT_FAILURE);
-    }
-
-    /* Cache parametres. */
-    std::list<int> associativities = { 1, 2, 4, 8, 16 };
-    std::list<int> cache_sizes = { 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024 };
-    uint32 addr; // storage for address
-
-    for ( auto associativity : associativities)
-    {
-        for ( auto cache_size : cache_sizes) // by cache size
-        {
-            CacheTagArray cta( 1024 * cache_size, associativity);
-            uint64 hit = 0, miss = 0;
-            while ( file_in >> hex >> addr) // while file contains addresses
-            {
-                if ( cta.read( addr).first) // hit
-                {
-                    hit++;
-                } else // miss
-                {
-                    miss++;
-                    cta.write( addr); // load to the cache
-                }
-            }
-            file_in.clear(); // reset "EOF" flag on file stream
-            file_in.seekg( ifstream::beg); // set file pointer to the beginning
-            double rate = 1.0 * miss / ( hit + miss);
-            file_out << rate << ", ";
-            file_out.flush(); // immidiate print to the file (to avoid delays)
-        }
-        file_out << std::endl;
-    }
-    /* Same as previous for full-associative cache. */
-    for ( auto cache_size : cache_sizes)
-    {
-        CacheTagArray cta( 1024 * cache_size, 1024 * cache_size / 4);
```

Comment:
That was deleted with nothing in return.

Reply:
I have said in the pull request description that there are some tests to be created so miss_rate_sim converted to GTest is one of them. I have planned to use it as the main and the only test for key methods like `write' and `read_no_touch`. However, I thought it would be enough to have bpu tests and LRUTagCache tests in the initial version.